src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2509,5 +2512,585 @@ uint8_t RadioSpinel<InterfaceType, ProcessContextType>::GetCslUncertainty(void)
+
+        otLogDebgPlat("%s ...", buf);
+        otLogDebgPlat(" txRequest:%u", metrics.mNumTxRequest);
+        otLogDebgPlat(" txGrantImmediate:%u", metrics.mNumTxGrantImmediate);
+        otLogDebgPlat(" txGrantWait:%u", metrics.mNumTxGrantWait);
+        otLogDebgPlat(" txGrantWaitActivated:%u", metrics.mNumTxGrantWaitActivated);
+        otLogDebgPlat(" txGrantWaitTimeout:%u", metrics.mNumTxGrantWaitTimeout);
+        otLogDebgPlat(" txGrantDeactivatedDuringRequest:%u", metrics.mNumTxGrantDeactivatedDuringRequest);
+        otLogDebgPlat(" txDelayedGrant:%u", metrics.mNumTxDelayedGrant);
+        otLogDebgPlat(" avgTxRequestToGrantTime:%u", metrics.mAvgTxRequestToGrantTime);
+        otLogDebgPlat(" rxRequest:%u", metrics.mNumRxRequest);
+        otLogDebgPlat(" rxGrantImmediate:%u", metrics.mNumRxGrantImmediate);
+        otLogDebgPlat(" rxGrantWait:%u", metrics.mNumRxGrantWait);
+        otLogDebgPlat(" rxGrantWaitActivated:%u", metrics.mNumRxGrantWaitActivated);
+        otLogDebgPlat(" rxGrantWaitTimeout:%u", metrics.mNumRxGrantWaitTimeout);
+        otLogDebgPlat(" rxGrantDeactivatedDuringRequest:%u", metrics.mNumRxGrantDeactivatedDuringRequest);
+        otLogDebgPlat(" rxDelayedGrant:%u", metrics.mNumRxDelayedGrant);
+        otLogDebgPlat(" avgRxRequestToGrantTime:%u", metrics.mAvgRxRequestToGrantTime);
+        otLogDebgPlat(" rxGrantNone:%u", metrics.mNumRxGrantNone);
+        otLogDebgPlat(" stopped:%u", metrics.mStopped);
+
+        start = buf;
+        start += snprintf(start, static_cast<size_t>(end - start), " grantGlitch:%u", metrics.mNumGrantGlitch);
+    }
+    break;
+
+    case SPINEL_PROP_MAC_SCAN_MASK:
+    {
+        constexpr uint8_t kNumChannels = 16;
+        uint8_t           channels[kNumChannels];
+        spinel_size_t     size;
+
+        unpacked = spinel_datatype_unpack(data, len, SPINEL_DATATYPE_DATA_S, channels, &size);
+        VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
+        start += snprintf(start, static_cast<size_t>(end - start), ", channels:");
+
+        for (uint8_t i = 0; i < size; i++)
+        {
+            start += snprintf(start, static_cast<size_t>(end - start), "%u ", channels[i]);
+        }
+    }
+    break;
+
+    case SPINEL_PROP_RCP_ENH_ACK_PROBING:
+    {
+        uint16_t saddr;
+        uint8_t  m8[OT_EXT_ADDRESS_SIZE];
+        uint8_t  flags;
+
+        unpacked = spinel_datatype_unpack(
+            data, len, SPINEL_DATATYPE_UINT16_S SPINEL_DATATYPE_EUI64_S SPINEL_DATATYPE_UINT8_S, &saddr, m8, &flags);
+
+        VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
+        start += snprintf(start, static_cast<size_t>(end - start),
+                          ", saddr:%04x, extaddr:%02x%02x%02x%02x%02x%02x%02x%02x, flags:0x%02x", saddr, m8[0], m8[1],
+                          m8[2], m8[3], m8[4], m8[5], m8[6], m8[7], flags);
+    }
+    break;
+    }
+
+exit:
+
+    if (otLoggingGetLevel() >= OT_LOG_LEVEL_DEBG)
```

Comment:
Do we need this `if` check? If we need it here, then do we need it also above when we do call `otLogDebgPlat()` (to log part of the info)?

I think we may not need this check.

Reply:
Agree, I have removed the check.